pkg/KubeArmorOperator/internal/controller/cluster.go

```diff
@@ -460,6 +460,7 @@ func (clusterWatcher *ClusterWatcher) WatchConfigCrd() {
 						UpdatedKubearmorRelayEnv(&cfg.Spec)
 						UpdatedSeccomp(&cfg.Spec)
 						UpdateRecommendedPolicyConfig(&cfg.Spec)
+						UpdateControllerPort(&cfg.Spec)
```

Comment:
add this check to update event as well and if changes detected update port to the controller deployment.